containers/go-postgres/.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,55 @@
+version: '3.8'
+
+volumes:
+  postgres-data:
+
+services:
+  app:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+      args:
+        # [Choice] Go version 1, 1.16, 1.17
+        # Append -bullseye or -buster to pin to an OS version.
+        # Use -bullseye variants on local arm64/Apple Silicon.
+        VARIANT: 1-bullseye
+        # Options
+        NODE_VERSION: "lts/*"
+    env_file:
+        # Ensure that the variables in .env match the same variables in devcontainer.json
+        - .env
+
+    # Security Opt and cap_add allow for C++ based debuggers to work. 
+    # See `runArgs`: https://github.com/Microsoft/vscode-docs/blob/main/docs/remote/devcontainerjson-reference.md
+#     security_opt:
+#       - seccomp:unconfined
+#     cap_add:
+#       - SYS_PTRACE
+
+    volumes:
+      - ..:/workspace:cached
+      
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+
+    # Comment out the next line to use a root user for all processes.
+    user: vscode
```

Comment:
The pattern we tend to use is to leave the container startup running as root and connect as non-root using the "remoteUser" property in devcontainer.json. If you don't entrypoints that need to start up a process end up needing to use sudo which isn't great either.  This way you can uninstall sudo and things function as expected.

Reply:
Fixed